lib/resolver.c

```diff
@@ -197,7 +198,9 @@ void resolver_resolve(struct resolver_query *query, int af,
 		zlog_debug("[%p] Resolving '%s'", query, hostname);
 
 	query->callback = callback;
+	vrf_switch_to_netns(vrf_id);
```

Comment:
This is insufficient. DNS resolution uses `/etc/resolv.conf` which is bind-mounted in a mount namespace when needed. This PR does not apply the mount namespace, so you end up using a netns without the matching DNS config.

Additionally, I'm not sure this feature is always desirable. There are use cases where some VRF may not have access to a DNS cache set up, e.g. because it is an "outside" VRF and firewall rules prevent it from accessing servers.

Reply:
it is up to the user to configure its setting appropriately. I agree with you.
/etc/resolv.conf may be common to all vrfs... or not.
and if not, then it makes sense to have a per-vrf behaviour.

in the case vrf backend, this makes sense.